index.bs

```diff
@@ -1130,6 +1156,392 @@ The following extended attributes are applicable to interfaces:
+          readonly attribute unsigned short entryType;
+          // ...
+        };
+
+        mixin Observable {
+          void addEventListener(DOMString type,
+                                EventListener listener,
+                                boolean useCapture);
+          // ...
+        };
+
+        Entry includes Observable;
+    </pre>
+
+    An ECMAScript implementation would thus have an “addEventListener”
+    property in the prototype chain of every <code class="idl">Entry</code>:
+
+    <pre highlight="js">
+        var e = getEntry();          // Obtain an instance of Entry.
+        typeof e.addEventListener;   // Evaluates to "function".
+    </pre>
+
+</div>
+
+
+<div class="note">
+
+    Mixins allow the sharing of [=attributes=], [=constants=], and [=operations=]
+    across <em>multiple</em> [=interfaces=].
+    If you're only planning to extend a single interface,
+    you might consider using a [=partial interface=] instead.
+
+    For example, instead of:
+
+    <pre class="webidl">
+        mixin WindowSessionStorage {
+          readonly attribute Storage sessionStorage;
+        };
+        Window includes WindowSessionStorage;
+    </pre>
+
+    do:
+
+    <pre class="webidl">
+        partial Window {
+          readonly attribute Storage sessionStorage;
+        };
+    </pre>
+
+    Additionally, you can rely on extending [=mixins=] exposed by other specifications
+    to target common use cases, such as [=exposed|exposing=]
+    a set of [=attributes=], [=constants=], or [=operations=]
+    across both window and worker contexts.
+
+    For example, instead of the common but verbose:
+
+    <pre class="webidl">
+        mixin GlobalCrypto {
+          [Throws] readonly attribute Crypto crypto;
+        };
+
+        Window includes GlobalCrypto;
+        WorkerGlobalScope includes GlobalCrypto;
+    </pre>
+
+    you can extend the {{WindowOrWorkerGlobalScope}} [=mixin=] using a [=partial mixin|partial=]:
+
+    <pre class="webidl">
+        partial mixin WindowOrWorkerGlobalScope {
+          [Throws] readonly attribute Crypto crypto;
+        };
+    </pre>
+</div>
+
+<div data-fill-with="grammar-Partial"></div>
+
+<div data-fill-with="grammar-PartialDefinition"></div>
+
+<pre class="grammar" id="prod-Mixin">
+    Mixin :
+        "mixin" identifier Inheritance "{" MixinMembers "}" ";"
+</pre>
+
+<pre class="grammar" id="prod-PartialMixin">
+    PartialMixin :
+        "mixin" identifier "{" MixinMembers "}" ";"
+</pre>
+
+<pre class="grammar" id="prod-MixinMembers">
+    MixinMembers :
+        ExtendedAttributeList MixinMember MixinMembers
+        ε
+</pre>
+
+<pre class="grammar" id="prod-MixinMember">
+    MixinMember :
+        Const
+        RegularOperation
+        Stringifier
+        ReadOnly AttributeRest
+</pre>
+
+<pre class="grammar" id="prod-IncludesStatement">
+    IncludesStatement :
+        identifier "includes" identifier ";"
+</pre>
+
+<h3 id="idl-namespaces">Namespaces</h3>
```

Comment:
Maybe we can land a re-arranging commit separately to make the diff clearer.

Reply:
Done.